tests/karate/src/test/java/be/createElection/castVote.feature

```diff
@@ -0,0 +1,74 @@
+    * call read('classpath:be/utils/server.feature')
+    * call read('classpath:be/mockFrontEnd.feature')
+    * def castVoteId = 41
+    * string electionChannel = "/root/p_EYbHyMv6sopI5QhEXBf40MO_eNoq7V_LygBd4c9RA=/ZVxXK2QN60uCNxNsIzShYYQmtwGttWLpQPQapYCNg4g="
+    * string laoChannel = "/root/p_EYbHyMv6sopI5QhEXBf40MO_eNoq7V_LygBd4c9RA="
+
+  # Testing if after setting up a valid lao, subscribing to it, sending a catchup
+  # creating a valid election setup that casting a valid vote succeeds
+  Scenario: Casting a valid vote on a started election
+    Given string castVoteData = read('classpath:data/election/data/castVote/valid_cast_vote_data.json')
+    And string castVote = converter.publishМessageFromData(castVoteData, castVoteId, electionChannel)
+    * call read('classpath:be/utils/simpleScenarios.feature@name=election_setup')
+    And eval frontend.send(castVote)
+    * json cast_vote = frontend_buffer.takeTimeout(timeout)
+    Then match cast_vote contains deep {jsonrpc: '2.0', id: '#(castVoteId)', result: 0}
+
+  Scenario: Casting a vote on a lao channel should return an error
+    Given string castVoteData = read('classpath:data/election/data/castVote/valid_cast_vote_2_data.json')
+    And string castVote = converter.publishМessageFromData(castVoteData, castVoteId, laoChannel)
+    * call read('classpath:be/utils/simpleScenarios.feature@name=election_setup')
+    And eval frontend.send(castVote)
+    * json cast_vote = frontend_buffer.takeTimeout(timeout)
+    Then match cast_vote contains deep {jsonrpc: '2.0', id: '#(castVoteId)', error: {code: -4, description: '#string'}}
+
+
+  Scenario: Casting a valid vote on non existent election should return an error
+    Given string badCastVoteData = read('classpath:data/election/data/castVote/bad_cast_vote_invalid_election_id_data.json')
+    And string badCastVote = converter.publishМessageFromData(badCastVoteData, castVoteId, electionChannel)
+    * call read('classpath:be/utils/simpleScenarios.feature@name=election_setup')
+    And eval frontend.send(badCastVote)
+    * json cast_vote = frontend_buffer.takeTimeout(timeout)
+    Then match cast_vote contains deep {jsonrpc: '2.0', id: '#(castVoteId)', error: {code: -4, description: '#string'}}
+
+  Scenario: Casting a valid vote with wrong vote id should return an error
+    Given string badCastVoteData = read('classpath:data/election/data/castVote/bad_cast_vote_invalid_vote_id_data.json')
+    And string badCastVote = converter.publishМessageFromData(badCastVoteData, castVoteId, electionChannel)
+    * call read('classpath:be/utils/simpleScenarios.feature@name=election_setup')
+    And eval frontend.send(badCastVote)
+    * json cast_vote = frontend_buffer.takeTimeout(timeout)
+    Then match cast_vote contains deep {jsonrpc: '2.0', id: '#(castVoteId)', error: {code: -4, description: '#string'}}
+
+  Scenario: Casting a valid vote too late should return an error
```

Comment:
With the changes in the protocol about the election end, is this still relevant ?

Reply:
Maybe this needs a little more discussion, since now there is technically no end time to an election because the organizer decides when the end will happen, but casting a vote after this end should still be rejected by the backend.

Reply:
True, I meant about the end time being used

Reply:
Yeah I agree completely this test as it is right now does not make sense. It should be changed to address the election end change in the protocol.